wurst/systems/spawns/ThiefBushWaterfall.wurst

```diff
@@ -1,24 +1,29 @@
 package ThiefBushWaterfall
 import HashMap
+import JumpSystem
+import TerrainUtils
+import ErrorHandling
 
 @configurable let THIEF_BUSH_WATERFALL_HEIGHT = 140.0
 @configurable let THIEF_BUSH_WATERFALL_DURATION = 1.0
 @configurable let THIEF_BUSH_WATERFALL_GRAVITY = 50.0
 
 let entranceToExitMap = new HashMap<region, rect>()
 
-function waterfallEntrance() returns boolean
+function waterfallEntrance()
     let hero = GetEnteringUnit()
     if hero.isType(UNIT_TYPE_GROUND)
         let exit = entranceToExitMap.get(GetTriggeringRegion())
-        let dest = exit.randomPoint()
-        // hero.pause()
-        // s__Jump_start(hero, dest.x, dest.y, THIEF_BUSH_WATERFALL_HEIGHT, THIEF_BUSH_WATERFALL_DURATION, THIEF_BUSH_WATERFALL_GRAVITY)
-    return false
-
-function waterfallExit() returns boolean
-    // EVENT_JUMP_UNIT.unpause()
-    return false
+        var dest = exit.randomPoint()
+        for i = 1 to 15

```

Comment:
This loop feels clunky. Also it looks like on the 15th try, if the point is not walkable then it will print an error but still make you jump there.

Reply:
Error stops the execution on the thread